aws/flatcar-linux/kubernetes/cl/controller.yaml.tmpl

```diff
@@ -214,6 +200,16 @@ storage:
           kind: KubeletConfiguration
           cgroupDriver: "$${docker_cgroup_driver}"
           EOF
+    - path: /etc/kubernetes/kubelet-wrapper
+      filesystem: root
+      mode: 0744
+      contents:
+        inline: ${kubelet_wrapper_inline}
```

Comment:
As the AWS Ignition size is a problem, I think the `*-wrapper` files can be just copied via SSH as done with `/etc/kubernetes/kubeconfig`.

Reply:
Ssh doesn't work for new nodes added by the AWS autoscaler. It kind of works for the controller node because it currently doesn't get scaled up much, but the same kubelet wrapper is needed on worker nodes where autoscaling is used more often.

I think we can use a private S3 bucket that we would provision by the same Terraform for AWS. And for packet, we have enough room for the user data (there is no limit at 16kb)

Reply:
Adding to what @alban said, I think we should minimize our dependence on SSH file copies. This mechanism was inherited from Typhoon, and the reason Typhoon chose to copy files over SSH rather than inject them via a cloud-native mechanism (userdata) is because in the Typhoon design the purpose was to have exactly the same installation process for all supported platforms. SSH was chosen because it is ubiquitous.

That said, copying files over SSH is less stable than using userdata (because it isn't atomic, unlike the userdata approach where if the machine got created it's more or less guaranteed the file will be created, too). Since stability and reproducibility are already problematic in Lokomotive, it should be our goal to *mitigate* such things rather than add more of them.

Reply:
Also, the problem is not only autoscaling, it's also **self-healing**: when a node gets deleted, we want it to be re-created automatically if the platform allows that. On AWS/Azure/GCP this should already work out of the box because we are utilizing ASGs or equivalents. on Packet it's a problem yet to be solved.